### PR TITLE
Various userland quality of life improvements

### PR DESCRIPTION
**Pull request by cstanfill**
Got rid of the DEBUG macro defined in every single test. Ported printf over and some basic stdlib functions.

### CODE REVIEW

File: userland/test-disk/main.c [hunk 1]
```diff
@@ -1,26 +1,13 @@
...
-  j[7] = 0;
-  k[0] = 'G';
-  k[1] = 'o';
-  k[2] = 'o';
-  k[3] = 'd';
-  k[4] = 'b';
-  k[5] = 'y';
-  k[6] = 'e';
-  k[7] = 0;
+  strncpy(j, "Hello!!", sizeof("Hello!!"));
```
Comment (gsilvis) [inline userland/test-disk/main.c]: strcpy would be even more convenient

File: userland/test-disk/main.c [hunk 2]
```diff
@@ -29,20 +16,8 @@ void main (void) {
...
-  if (k[5] != '!')  return;
-  if (k[6] != '!')  return;
-  if (j[0] != 'G')  return;
-  if (j[1] != 'o')  return;
-  if (j[2] != 'o')  return;
-  if (j[3] != 'd')  return;
-  if (j[4] != 'b')  return;
-  if (j[5] != 'y')  return;
-  if (j[6] != 'e')  return;
+  if (strncmp(k, "Hello!!", strlen("Hello!!"))) return;
```
Comment (gsilvis) [inline userland/test-disk/main.c]: ditto, strcmp rather than strncmp

File: userland/test-disk/main.c (hunk 1 above)
Comment (cstanfill) [reply userland/test-disk/main.c]: Done

File: userland/test-disk/main.c (hunk 2 above)
Comment (cstanfill) [reply userland/test-disk/main.c]: Done

**Comment by cstanfill** [comment]
Rebased and modified older commits to include your new tests.

**Pull request merged**